index.js

```diff
@@ -128,7 +112,8 @@ module.exports = {
       this.ui.writeWarnLine('[ember-fetch] Could not find `ember-cli-babel` addon, opting out of transpilation!')
     }
 
-    const preferNative = this.buildConfig.preferNative;
+    const app = this._findApp();
```

Comment:
Can we store `app` as `this.app` to avoid calling `_findApp()` twice?

Reply:
That it basically the whole issue that this PR is trying to fix: `this` is not the same in these different calls, at least not when run in an engine with nested dependencies.

I'm not sure how this is _supposed_ to work, but in my app with an engine, if I adjust the code like this:

```js
included() {
  this.__tid = Math.random();
  console.log('included', this.__tid);
  // ...
},
treeForVendor() {
  console.log('treeForVendor', this.__tid);
  // ...
}
```

I get this output:

```
included 0.47868310290718585
included 0.4535138056640218
treeForVendor undefined
treeForVendor undefined
```
(This is in my app with ember-cli 3.4)

Reply:
Thanks for the explanation!